lib/workers/api_use.js

```diff
@@ -66,70 +66,86 @@ function postProcess(parsedFiles, filenames, preProcess, packageInfos, source, t
 
     parsedFiles.forEach(function(parsedFile, parsedFileIndex) {
         parsedFile.forEach(function(block) {
-            if ( ! block.local[target])
-                return;
+            var loopCounter = 0;     //add a loop counter to have a break condition when the recursion depth exceed a predifined limit
+	    while (block.local[target]) {
+	        if (loopCounter > 10) {
+	            throw new WorkerError('recursion depth exceeds limit with @apiUse',
+		        filenames[parsedFileIndex],
+		        block.index,
+		        messages.common.element,
+		        messages.common.usage,
+		        messages.common.example,
+		        [
+			   { 'Groupname': name }
+		        ]
+                    );
+	        }
+                
+                //create a copy of the elements for save iterating of the elements
+                const blockClone = [...block.local[target]];
 
-            block.local[target].forEach(function(definition) {
-                var name = definition.name;
-                var version = block.version || packageInfos.defaultVersion;
+                // remove unneeded target before starting the loop, to allow a save insertion of new elements
+                // TODO: create a cleanup filter
```

Comment:
could you expand on that? It's not great to add a TODO ;)

Reply:
This is not from me. I only moved it to another place

Reply:
In the connected ticket you can see the changes i made in a better overview. Githubs pullreuest didnt resolve the extra space i add for the new while loop